src/main/java/org/swisspush/redisques/handler/GetQueueItemsCountHandler.java

```diff
@@ -22,15 +23,21 @@ public class GetQueueItemsCountHandler implements Handler<AsyncResult<Response>>
 
     private static final Logger log = getLogger(GetQueueItemsCountHandler.class);
     private final Message<JsonObject> event;
+    private final QueueConfiguration queueConfiguration;
 
-    public GetQueueItemsCountHandler(Message<JsonObject> event) {
+    public GetQueueItemsCountHandler(Message<JsonObject> event, QueueConfiguration queueConfiguration) {
         this.event = event;
+        this.queueConfiguration = queueConfiguration;
     }
 
     @Override
     public void handle(AsyncResult<Response> reply) {
-        if(reply.succeeded()){
+        if (reply.succeeded()) {
             Long queueItemCount = reply.result().toLong();
+            if (queueConfiguration != null && queueConfiguration.getMaxQueueEntries() > 0) {
```

Comment:
Again, reading the queue should return the truth.

Reply:
emm, maybe there have other way, I don't need to lie the results.

if Queue is QueueState.CONSUMING: trim as what implemented now
if Queue is QueueState.READY: trim when add queue item as before

Reply:
if modify the when add queue item will back to my first version, that will have problem when queue also consuming, the lpop at readQueue, may removed wrong Item which is not consumed, Let me update code which I think is safe, let you check again.